README.md

```diff
@@ -1,18 +1,66 @@
-# `android-ndk`: Rust bindings of the Android NDK
+# Rust on Android
 
-[![Build Status](https://travis-ci.org/mb64/android-ndk-rs.svg?branch=master)](https://travis-ci.org/mb64/android-ndk-rs)
-[![Crates.io Status](https://meritbadge.herokuapp.com/android-ndk-sys)](https://crates.io/crates/android-ndk-sys)
-[![Docs.rs Status](https://docs.rs/android-ndk-sys/badge.svg)](https://docs.rs/android-ndk-sys)
-[![Crates.io Status](https://meritbadge.herokuapp.com/android-ndk)](https://crates.io/crates/android-ndk)
-[![Docs.rs Status](https://docs.rs/android-ndk/badge.svg)](https://docs.rs/android-ndk)
+ - Raw FFI bindings to the NDK ![ndk-sys-docs][ndk-sys-badge]
+ - Safe abstraction of the bindings ![ndk-docs][ndk-badge]
+ - Startup code ![ndk-glue-docs][ndk-glue-badge]
+ - Everything for building apk's ![ndk-build-docs][ndk-build-badge]
+ - Build tool ![cargo-apk-docs][cargo-apk-badge]
 
-This is a work in progress at the moment.
+## Hello world
 
-`android-ndk-sys` contains the raw FFI bindings, pre-generated from NDK r20, and `android-ndk`
-provides a safe API over it.
+```toml
+[lib]
+crate-type = ["cdylib"]
+```
 
-Other helpful crates for Android:
+```rust
+#[cfg(target_os = "android")]
+#[no_mangle]
+pub unsafe extern "C" fn ANativeActivity_onCreate(
+    activity: *mut std::os::raw::c_void,
+    saved_state: *mut std::os::raw::c_void,
+    saved_state_size: usize,
+) {
+    ndk_glue::init(
+        activity as _,
+        saved_state as _,
+        saved_state_size as _,
+        app_main,
+    );
+}
 
- * [`jni`](https://crates.io/crates/jni), JNI bindings for Rust
- * [`android_logger`](https://crates.io/crates/android_logger) and [`ndk-logger`](https://crates.io/crates/ndk-logger),
-   Android backends for the `log` crate
+pub fn app_main() {
+    println!("hello world");
+}
+```
+
+```sh
+cargo install cargo-apk
+cargo apk run
```

Comment:
Alright, so yesterday I tried this, which didn't work:
```
  ~/Do…ts/gfx/eco3/glutin                                                                                                                           SIGINT(2) ↵  69%  
  12:39  17.02.20  cargo apk run --example window
   Compiling glutin v0.23.0-alpha1 (/home/gentz/Documents/gfx/eco3/glutin)
    Finished dev [unoptimized + debuginfo] target(s) in 6.00s
Error: Path "/home/gentz/Documents/gfx/eco3/glutin/target/aarch64-linux-android/debug/examples/libwindow.so" doesn't exist.
```

You mention it needs to be cdylib, so I tried this.

```
  ~/Do…ts/gfx/eco3/glutin                                                                                                                      1 ↵  6.41981s   69%  
  12:39  17.02.20  RUSTFLAGS="--crate-type cdylib" cargo apk run --example window  
   Compiling cfg-if v0.1.10
   Compiling lazy_static v1.4.0
   Compiling scopeguard v1.1.0
   Compiling adler32 v1.0.4
error: `#[panic_handler]` function required, but not found

error: language item required, but not found: `eh_personality`

error: aborting due to 2 previous errors

error: could not compile `cfg-if`.
warning: build failed, waiting for other jobs to finish...
error: `#[panic_handler]` function required, but not found

error: language item required, but not found: `eh_personality`

error: aborting due to 2 previous errors

error: could not compile `scopeguard`.
warning: build failed, waiting for other jobs to finish...
error: build failed
Error: Command 'cargo build --target aarch64-linux-android --example window' had a non-zero exit code.
```

I did some googling, and this gets rid of the `eh_personality` error:
```
  ~/Do…ts/gfx/eco3/glutin                                                                                                                      1 ↵  1.15127s   69%  
  12:40  17.02.20  RUSTFLAGS="--crate-type cdylib -C panic=abort" cargo apk run --example window
   Compiling cfg-if v0.1.10
   Compiling lazy_static v1.4.0
   Compiling scopeguard v1.1.0
   Compiling adler32 v1.0.4
error: `#[panic_handler]` function required, but not found

error: aborting due to previous error

error: could not compile `cfg-if`.
warning: build failed, waiting for other jobs to finish...
error: `#[panic_handler]` function required, but not found

error: aborting due to previous error

error: could not compile `scopeguard`.
warning: build failed, waiting for other jobs to finish...
error: build failed
Error: Command 'cargo build --target aarch64-linux-android --example window' had a non-zero exit code.
```

Implementing panic handler on a function wouldn't fix that error, though, so this readme leaves me kind of stumped :/

Reply:
This works for me:
```diff
diff --git a/Cargo.toml b/Cargo.toml
index 3898294..b2c6232 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -74,3 +74,7 @@ glutin_glx_sys = { version = "0.2.0", path = "../glutin_sys/glutin_glx_sys" }
 parking_lot = "0.10.0"
 glutin_x11_sym = { version = "0.1.0", path = "../glutin_x11_sym" }
 x11-dl = "2.18.4"
+
+[[example]]
+name = "window"
+crate-type = ["cdylib"]
```

Reply:
The rustflags env the way you set it gives me the same error

Reply:
The problem seems to be that it tries to compile all crates as cdylib, when only the example needs to be cdylib. Trying to compile a proc-macro as cdylib is no good.

Reply:
also note that it'll crash in any case. we could add support to the ios-example and rename it to mobile-example?

Reply:
Well, that sucks. I guess we could follow this suggestion I stole off of reddit: https://www.reddit.com/r/rust/comments/64hlfa/how_to_compile_a_crate_as_dynamic_library/dg2rq64/
I tried the following, but:
```
$ rm android -rf; mkdir android; cat Cargo.toml | sed "s/\.\.\//\.\.\/\.\.\//g" > android/Cargo.toml; ls examples/ | grep rs | awk '{sub(/\..*/,"",$1); print "\n[[example]]\nname = \"" $0 "\"\ncrate-type = [\"cdylib\"]\npath = \"../examples/" $0 ".rs\""}' >> android/Cargo.toml; printf '\n[lib]\npath = "../src/lib.rs"\n' >> android/Cargo.toml; cargo apk run --example window --manifest-path android/Cargo.toml
ERROR: unable to open 'android/target/debug/apk/examples/window-unaligned.apk' as Zip file for writing
    Updating crates.io index
error: no matching package named `ndk` found
location searched: registry `https://github.com/rust-lang/crates.io-index`
required by package `winit v0.21.0 (/home/gentz/Documents/gfx/eco3/winit)`
    ... which is depended on by `glutin v0.23.0-alpha1 (/home/gentz/Documents/gfx/eco3/glutin/android)`
Error: Command 'cargo build --target armv7-linux-androideabi --example window --manifest-path android/Cargo.toml' had a non-zero exit code.
```

```
$ cat android/Cargo.toml         
[package]
name = "glutin"
version = "0.23.0-alpha1"
authors = ["The glutin contributors", "Pierre Krieger <pierre.krieger1708@gmail.com>"]
description = "Cross-platform OpenGL context provider."
keywords = ["windowing", "opengl"]
license = "Apache-2.0"
readme = "../../README.md"
repository = "https://github.com/rust-windowing/glutin"
documentation = "https://docs.rs/glutin"
edition = "2018"
build = "build.rs"

[package.metadata.docs.rs]
features = ["serde_feature"]

[features]
serde_feature = ["winit_types/serde_feature"]

[dependencies]
lazy_static = "1.4.0"
winit_types = { version = "0.1.0", path = "../../winit_types" }
glutin_interface = { version = "0.1.0", path = "../../glutin_interface" }
log = "0.4.8"
libloading = "0.5.2"

[build-dependencies]
gl_generator = "0.14.0"

[dev-dependencies]
winit = { path = "../../winit" }
takeable-option = "0.5.0"
image = "0.22.4"
simple_logger = "1.4.0"

[target.'cfg(all(any(target_os = "linux", target_os = "freebsd", target_os = "dragonfly", target_os = "netbsd", target_os = "openbsd"), target_arch="x86_64"))'.dev-dependencies]
gbm = { path = "../../gbm.rs" }
drm = "0.3.4"

[target.'cfg(target_os = "android")'.dependencies]
glutin_egl_sys = { version = "0.1.4", path = "../../glutin_sys/glutin_egl_sys" }
parking_lot = "0.10.0"

[target.'cfg(any(target_os = "ios", target_os = "macos"))'.dependencies]
objc = "0.2.7"
glutin_gles2_sys = { version = "0.1.3", path = "../../glutin_sys/glutin_gles2_sys" }

[target.'cfg(target_os = "macos")'.dependencies]
cgl = "0.3.2"
cocoa = "0.20.0"
core-foundation = "0.7.0"
core-graphics = "0.19.0"

[target.'cfg(target_os = "windows")'.dependencies]
libloading = "0.5.2"
glutin_wgl_sys = { version = "0.1.3", path = "../../glutin_sys/glutin_wgl_sys" }
glutin_egl_sys = { version = "0.1.5", path = "../../glutin_sys/glutin_egl_sys" }
parking_lot = "0.10.0"

[target.'cfg(target_os = "windows")'.dependencies.winapi]
version = "0.3.8"
features = [
    "winnt",
    "winuser",
    "wingdi",
    "libloaderapi",
]

[target.'cfg(any(target_os = "linux", target_os = "freebsd", target_os = "dragonfly", target_os = "netbsd", target_os = "openbsd"))'.dependencies]
glutin_osmesa_sys = { version = "0.1.0", path = "../../glutin_sys/glutin_osmesa_sys" }
wayland-client = { version = "0.23", features = ["egl", "dlopen"] }
glutin_egl_sys = { version = "0.1.4", path = "../../glutin_sys/glutin_egl_sys" }
glutin_glx_sys = { version = "0.2.0", path = "../../glutin_sys/glutin_glx_sys" }
parking_lot = "0.10.0"
glutin_x11_sym = { version = "0.1.0", path = "../../glutin_x11_sym" }
x11-dl = "2.18.4"

[[example]]
name = "damage"
crate-type = ["cdylib"]
path = "../examples/damage.rs"

[[example]]
name = "egldevice"
crate-type = ["cdylib"]
path = "../examples/egldevice.rs"

[[example]]
name = "fullscreen"
crate-type = ["cdylib"]
path = "../examples/fullscreen.rs"

[[example]]
name = "gbm"
crate-type = ["cdylib"]
path = "../examples/gbm.rs"

[[example]]
name = "headless"
crate-type = ["cdylib"]
path = "../examples/headless.rs"

[[example]]
name = "make_current_rw"
crate-type = ["cdylib"]
path = "../examples/make_current_rw.rs"

[[example]]
name = "multiwindow"
crate-type = ["cdylib"]
path = "../examples/multiwindow.rs"

[[example]]
name = "pixmap"
crate-type = ["cdylib"]
path = "../examples/pixmap.rs"

[[example]]
name = "sharing"
crate-type = ["cdylib"]
path = "../examples/sharing.rs"

[[example]]
name = "transparent"
crate-type = ["cdylib"]
path = "../examples/transparent.rs"

[[example]]
name = "vsync"
crate-type = ["cdylib"]
path = "../examples/vsync.rs"

[[example]]
name = "window"
crate-type = ["cdylib"]
path = "../examples/window.rs"

[lib]
path = "../src/lib.rs"
```

Reply:
There may be a way `cargo apk -- rustc --example window -- --crate-type cdylib`. This doesn't currently create an apk but it compiles the example.

Reply:
Hmmm, but I was hoping to make an apk.

Reply:
Alright, so I applied onto winit:
```patch
diff --git a/Cargo.toml b/Cargo.toml
index a64a0583..e07eef7c 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -35,9 +35,9 @@ image = "0.22.4"
 simple_logger = "1.4.0"
 
 [target.'cfg(target_os = "android")'.dependencies]
-ndk = "0.1.0"
-ndk-glue = "0.1.0"
-ndk-sys = "0.1.0"
+ndk = { version = "0.1.0", path = "../android-ndk-rs/ndk" }
+ndk-glue = { version = "0.1.0", path = "../android-ndk-rs/ndk-glue" }
+ndk-sys = { version = "0.1.0", path = "../android-ndk-rs/ndk-sys" }
 
 [target.'cfg(target_os = "ios")'.dependencies]
 objc = "0.2.7"
```
Then I ran my long zsh one-liner:
```
$ rm android -rf; mkdir android; cat Cargo.toml | sed "s/\.\.\//\.\.\/\.\.\//g;s/build\.rs/\.\.\/build\.rs/" > android/Cargo.toml; ls examples/ | grep rs | awk '{sub(/\..*/,"",$1); print "\n[[example]]\nname = \"" $0 "\"\ncrate-type = [\"cdylib\"]\npath = \"../examples/" $0 ".rs\""}' >> android/Cargo.toml; printf '\n[lib]\npath = "../src/lib.rs"\n' >> android/Cargo.toml; cargo apk run --example window --manifest-path android/Cargo.toml
ERROR: unable to open 'android/target/debug/apk/examples/window-unaligned.apk' as Zip file for writing
    Updating crates.io index
   Compiling libc v0.2.66
   Compiling cfg-if v0.1.10
   Compiling autocfg v1.0.0
   Compiling log v0.4.8
   Compiling lazy_static v1.4.0
   Compiling autocfg v0.1.7
   Compiling proc-macro2 v0.4.30
   Compiling unicode-xid v0.1.0
   Compiling semver-parser v0.7.0
   Compiling syn v0.15.44
   Compiling proc-macro2 v1.0.8
   Compiling scopeguard v1.1.0
   Compiling khronos_api v3.1.0
   Compiling serde v1.0.104
   Compiling unicode-xid v0.2.0
   Compiling byteorder v1.3.4
   Compiling xml-rs v0.8.0
   Compiling syn v1.0.14
   Compiling rayon-core v1.7.0
   Compiling bitflags v1.2.1
   Compiling num-derive v0.2.5
   Compiling cc v1.0.50
   Compiling adler32 v1.0.4
   Compiling crc32fast v1.2.0
   Compiling ndk-sys v0.1.0 (/home/gentz/Documents/gfx/eco3/android-ndk-rs/ndk-sys)
   Compiling winit_types v0.1.0 (/home/gentz/Documents/gfx/eco3/winit_types)
   Compiling smallvec v1.2.0
   Compiling lzw v0.10.0
   Compiling either v1.5.3
   Compiling jni-sys v0.3.0
   Compiling color_quant v1.0.1
   Compiling android_log-sys v0.1.2
   Compiling instant v0.1.2
   Compiling scoped_threadpool v0.1.9
   Compiling takeable-option v0.5.0
   Compiling num-traits v0.2.11
   Compiling num-integer v0.1.42
   Compiling num-iter v0.1.40
   Compiling num-rational v0.2.3
   Compiling semver v0.9.0
   Compiling crossbeam-utils v0.7.0
   Compiling crossbeam-epoch v0.8.0
   Compiling lock_api v0.3.3
   Compiling inflate v0.4.5
   Compiling glutin_interface v0.1.0 (/home/gentz/Documents/gfx/eco3/glutin_interface)
   Compiling libloading v0.5.2
   Compiling gif v0.10.3
   Compiling rustc_version v0.2.3
   Compiling num_cpus v1.12.0
   Compiling atty v0.2.14
   Compiling parking_lot_core v0.7.0
   Compiling time v0.1.42
   Compiling raw-window-handle v0.3.3
   Compiling quote v0.6.13
   Compiling memoffset v0.5.3
   Compiling gl_generator v0.14.0
   Compiling quote v1.0.2
   Compiling deflate v0.7.20
   Compiling colored v1.9.2
   Compiling parking_lot v0.10.0
   Compiling crossbeam-queue v0.2.1
   Compiling png v0.15.3
   Compiling glutin_egl_sys v0.1.6 (/home/gentz/Documents/gfx/eco3/glutin_sys/glutin_egl_sys)
   Compiling glutin v0.23.0-alpha1 (/home/gentz/Documents/gfx/eco3/glutin/android)
   Compiling chrono v0.4.10
   Compiling toml v0.5.6
   Compiling simple_logger v1.5.0
   Compiling crossbeam-deque v0.7.2
   Compiling derivative v1.0.3
   Compiling proc-macro-crate v0.1.4
   Compiling rayon v1.3.0
   Compiling num_enum_derive v0.4.2
   Compiling tiff v0.3.1
   Compiling num_enum v0.4.2
   Compiling ndk v0.1.0 (/home/gentz/Documents/gfx/eco3/android-ndk-rs/ndk)
   Compiling ndk-glue v0.1.0 (/home/gentz/Documents/gfx/eco3/android-ndk-rs/ndk-glue)
   Compiling winit v0.21.0 (/home/gentz/Documents/gfx/eco3/winit)
   Compiling jpeg-decoder v0.1.18
   Compiling image v0.22.5
warning: function is never used: `main`
  --> ../examples/window.rs:10:4
   |
10 | fn main() {
   |    ^^^^
   |
   = note: `#[warn(dead_code)]` on by default

    Finished dev [unoptimized + debuginfo] target(s) in 3m 54s
ERROR: failed opening/creating 'android/target/debug/apk/examples/window-unaligned.apk' as Zip file
Error: Command '/home/gentz/Android/Sdk/build-tools/29.0.3/aapt add android/target/debug/apk/examples/window-unaligned.apk lib/armeabi-v7a/libwindow.so' had a non-zero exit code.
```
Unfortunately, cargo-apk is outputting the apk to the wrong directory:

```
$ find target/ | grep "unal"
target/debug/apk/examples/window-unaligned.apk

$ find android/target | grep "unal"
```

Reply:
why is that the wrong directory? since it builds fat apk's (includes all archs in the same apk) it is not placed in the target directory. if building fat apks is a good idea or not is a different question, but it's cargo-apks original behaviour, so I kept it.

Reply:
working on it..

Reply:
I think this is resolved now, let me know if it works for you

Reply:
I get fetched both repos, git checkout'd the newest version and ran cargo install again. Still seams to be ignoring the set manifest path. 
```
  ~/Do…ts/gfx/eco3/glutin                                                                                                                    101 ↵  0.04256s   70%  
  15:48  17.02.20  cargo apk run --example window --manifest-path android/Cargo.toml --target aarch64-linux-android
ERROR: unable to open 'android/target/debug/apk/examples/window-unaligned.apk' as Zip file for writing
   Compiling glutin v0.23.0-alpha1 (/home/gentz/Documents/gfx/eco3/glutin/android)
    Finished dev [unoptimized + debuginfo] target(s) in 11.42s
ERROR: failed opening/creating 'android/target/debug/apk/examples/window-unaligned.apk' as Zip file
Error: Command '/home/gentz/Android/Sdk/build-tools/29.0.3/aapt add android/target/debug/apk/examples/window-unaligned.apk lib/arm64-v8a/libwindow.so' had a non-zero exit code.

  ~/Do…ts/gfx/eco3/glutin                                                                                                                     1 ↵  11.74608s   70%  
  15:48  17.02.20  find target android/target | grep window-u
target/debug/apk/examples/window-unaligned.apk
```

Reply:
Actually, it is not making an apk at all.
```
  ~/Do…ts/gfx/eco3/glutin                                                                                                                          0.04887s   70%  
  15:49  17.02.20  rm target -rf    

  ~/Do…ts/gfx/eco3/glutin                                                                                                                          0.25960s   70%  
  15:51  17.02.20  cargo apk run --example window --manifest-path android/Cargo.toml --target aarch64-linux-android
ERROR: unable to open 'android/target/debug/apk/examples/window-unaligned.apk' as Zip file for writing
   Compiling glutin v0.23.0-alpha1 (/home/gentz/Documents/gfx/eco3/glutin/android)
    Finished dev [unoptimized + debuginfo] target(s) in 10.09s
ERROR: failed opening/creating 'android/target/debug/apk/examples/window-unaligned.apk' as Zip file
Error: Command '/home/gentz/Android/Sdk/build-tools/29.0.3/aapt add android/target/debug/apk/examples/window-unaligned.apk lib/arm64-v8a/libwindow.so' had a non-zero exit code.

  ~/Do…ts/gfx/eco3/glutin                                                                                                                     1 ↵  10.47813s   70%  
  15:51  17.02.20  find target android/target | grep window-u
find: ‘target’: No such file or directory
```

Reply:
I'll look into it tomorrow, need to sleep...

Reply:
Does it still do that after I reverted the `cargo-rustc` hack?

Reply:
Still broken when using `--manifest-path`:
```
  ~/Do…ts/gfx/eco3/glutin                                                                                                                                       70%  
  17:14  18.02.20  find target android/target | grep window-u | xargs rm

  ~/Do…ts/gfx/eco3/glutin                                                                                                                          0.07800s   70%  
  17:14  18.02.20  find target android/target | grep window-u           

  ~/Do…ts/gfx/eco3/glutin                                                                                                                    0|1 ↵  0.03062s   70%  
  17:14  18.02.20  cargo apk run --example window --manifest-path android/Cargo.toml --target aarch64-linux-android
ERROR: unable to open 'android/target/debug/apk/examples/window-unaligned.apk' as Zip file for writing
   Compiling glutin v0.23.0-alpha1 (/home/gentz/Documents/gfx/eco3/glutin/android)
    Finished dev [unoptimized + debuginfo] target(s) in 10.30s
ERROR: failed opening/creating 'android/target/debug/apk/examples/window-unaligned.apk' as Zip file
Error: Command '/home/gentz/Android/Sdk/build-tools/29.0.3/aapt add android/target/debug/apk/examples/window-unaligned.apk lib/arm64-v8a/libwindow.so' had a non-zero exit code.

  ~/Do…ts/gfx/eco3/glutin                                                                                                                     1 ↵  10.75456s   70%  
  17:14  18.02.20  find target android/target | grep window-u

  ~/Do…ts/gfx/eco3/glutin                                                                                                                    0|1 ↵  0.03277s   70%  
  17:15  18.02.20  cargo apk run --example window --manifest-path android/Cargo.toml --target aarch64-linux-androi 
Error: Target is not supported.

  ~/Do…ts/gfx/eco3/glutin                                                                                                                      1 ↵  0.11674s   70%  
  17:15  18.02.20  cargo apk run --example window --manifest-path android/Cargo.toml                              
ERROR: unable to open 'android/target/debug/apk/examples/window-unaligned.apk' as Zip file for writing
   Compiling glutin v0.23.0-alpha1 (/home/gentz/Documents/gfx/eco3/glutin/android)
    Finished dev [unoptimized + debuginfo] target(s) in 11.00s
ERROR: failed opening/creating 'android/target/debug/apk/examples/window-unaligned.apk' as Zip file
Error: Command '/home/gentz/Android/Sdk/build-tools/29.0.3/aapt add android/target/debug/apk/examples/window-unaligned.apk lib/arm64-v8a/libwindow.so' had a non-zero exit code.

  ~/Do…ts/gfx/eco3/glutin                                                                                                                     1 ↵  11.54048s   70%  
  17:15  18.02.20  find target android/target | grep window-u                                                     

  ~/Do…ts/gfx/eco3/glutin                                                                                                                    0|1 ↵  0.02819s   70%  
  17:15  18.02.20  cd android/                                                      

  ~/Do…ts/gfx/eco3/gl…in/android                                                                                                                   0.00127s   70%  
  17:15  18.02.20  cargo apk run --example window                                   
   Compiling glutin v0.23.0-alpha1 (/home/gentz/Documents/gfx/eco3/glutin/android)
    Finished dev [unoptimized + debuginfo] target(s) in 10.11s
 'lib/arm64-v8a/libwindow.so'...
Verifying alignment of /home/gentz/Documents/gfx/eco3/glutin/android/target/debug/apk/examples/window.apk (4)...
      49 AndroidManifest.xml (OK - compressed)
     970 lib/arm64-v8a/libwindow.so (OK - compressed)
Verification succesful
Generating 2,048 bit RSA key pair and self-signed certificate (SHA256withRSA) with a validity of 10,000 days
	for: CN=Android Debug, O=Android, C=US
[Storing /home/gentz/.android/debug.keystore]

Warning:
The JKS keystore uses a proprietary format. It is recommended to migrate to PKCS12 which is an industry standard format using "keytool -importkeystore -srckeystore /home/gentz/.android/debug.keystore -destkeystore /home/gentz/.android/debug.keystore -deststoretype pkcs12".
error: no devices/emulators found
Performing Push Install
adb: error: failed to get feature set: no devices/emulators found
Error: Command '/home/gentz/Android/Sdk/platform-tools/adb install -r /home/gentz/Documents/gfx/eco3/glutin/android/target/debug/apk/examples/window.apk' had a non-zero exit code.
```

Reply:
The problem is that the path returned from cargo-subcommand to the `Cargo.toml` is relative to the `current_dir`, but the `aapt` command is executed from the `target/{profile}/apk` dir. Should be fixed now